app/Zone.php

```diff
@@ -186,17 +186,21 @@ public function records()
      */
     public function getNewSerialNumber(bool $force = false): int
     {
-        // Get current Zone serial number.
-        $currentSerial = $this->serial;
-
-        // We need a new one ONLY if there isn't pending changes.
-        if ($this->hasPendingChanges() && ! $force) {
-            return $currentSerial;
+        if (isset($this->serial)) {
```

Comment:
@pacoorozco This became a problem because I changed ZoneController::fillZoneFromRequest to call getNewSerialNumber() instead of assign $zone->serial = Zone::generateSerialNumber(). This introduced the case of ! isset($this->serial).
The reason for that change was that this assignment was re-setting and decrementing serial numbers when the zone was being saved.

I'm not too familiar with Laravel etc., I see right now you're doing:
$zone = factory(Zone::class)->create();
$expectedSerial = $zone->serial;
and that worked, so I suppose this creates a zone with a pre-seeded serial?
So the test you're asking for would be more like:
$zone = new Zone();
$zone->getNewSerialNumber();
$this->assertGreaterThan(0, $zone->serial);